src/main/java/uk/gov/hmcts/ccd/domain/service/search/filter/CaseFilterFactory.java

```diff
@@ -0,0 +1,9 @@
+package uk.gov.hmcts.ccd.domain.service.search.filter;
+
+import java.util.Optional;
+
+public interface CaseFilterFactory<T> {
```

Comment:
does this need to be a generic? I can't see this returning other types than a `Optional<QueryBuilder>`

Reply:
changed to non generic.